### PR TITLE
Generate random access tokens for Users

### PR DESCRIPTION
**Pull request by treiff**
@zspencer I gave this a shot.  Have a look.  

I couldn't easily think of a way to test if the `access_token` is non-retrievable.  However, I believe my spec covers the random generation aspect.

### CODE REVIEW

File: lib/user_inviter.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class UserInviter
   def invite(email)
-    user = User.create(email: email, access_token: "asdf")
-    "asdf"
+    user = User.create(email: email, access_token: User.generate_token)
+    user.access_token
```
Comment (zspencer) [inline lib/user_inviter.rb]: This is the encrypted access token, not the un-encrypted one. We need the unencrypted access token so we can hand it off to the user.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 class User < ActiveRecord::Base
-
-end
\ No newline at end of file
+  def self.generate_token
+    BCrypt::Password.create(SecureRandom.urlsafe_base64)
+  end
```
Comment (zspencer) [inline app/models/user.rb]: This is decent, but it means we have no way of knowing at the beginning what the unencrypted password is.

If we create the unencrypted password at the inviter level we can return it from the inviter and print it to the screen.

Then, we could use a similar process to how ActiveRecord's `has_secure_password` works to encrypt the access_token when we set it on the model. See: http://edgeapi.rubyonrails.org/classes/ActiveModel/SecurePassword/InstanceMethodsOnActivation.html and click the "show source" links.

File: spec/features/invite_users_spec.rb [hunk 3]
```diff
@@ -18,6 +20,16 @@
...
     end
+
+    it "generates a random access token" do
+      inviter = UserInviter.new()
+      result = inviter.invite("zee@example.com")
+      user = User.find_by(email: "zee@example.com")
+      user.update_attributes(access_token: User.generate_token)
+      updated_user = User.find_by(email: "zee@example.com")
+
+      expect(result).not_to eql updated_user.access_token
```
Comment (zspencer) [inline spec/features/invite_users_spec.rb]: This test has a lot of noise around it. If we want to demonstrate that it randomly generates access tokens, we could write a test like so:

```
inviter = UserInviter.new()
person_a_access_token = inviter.invite("person-a@example.com")
person_b_access_token = inviter.invite("person-b@example.com")
expect(person_a_access_token).not_to eql(person_b_access_token)
```

This doesn't really demonstrate that it's truly 100% random, but it's OK-ish for now. If we wanted to be pretty certain it's random we could generate a certain number of access tokens (Say, 5000) and confirm that none of them are the same as one another. 

An even better choice is to [default to using a uuid](http://en.wikipedia.org/wiki/Universally_unique_identifier) for the access token to have a really strong likelihood that the access token is unique.

Comment (treiff) [inline spec/features/invite_users_spec.rb]: Ok, Thanks.  That clears up some things.  I’ll revisit this.

**Pull request merged**